execution/relation.go

```diff
@@ -168,13 +174,74 @@ func (rel *Like) Apply(ctx context.Context, variables octosql.Variables, left, r
 			leftValue.Show(), rightValue.Show(), leftValue.GetType(), rightValue.GetType())
 	}
 
-	match, err := regexp.MatchString(rightValue.AsString(), leftValue.AsString())
+	patternString, err := likePatternToRegexp(rightValue.AsString())
+	if err != nil {
+		return false, errors.Wrapf(err, "couldn't transform LIKE pattern %v to regexp", patternString)
+	}
+
+	match, err := regexp.MatchString(patternString, leftValue.AsString())
 	if err != nil {
 		return false, errors.Wrapf(err, "couldn't match string in like relation with pattern %v", rightValue)
 	}
 	return match, nil
 }
 
+//we assume that the escape character is '\'
+func likePatternToRegexp(pattern string) (string, error) {
+	length := len(pattern)
+	if pattern[length-1] == LikeEscape {
+		return "", errors.New("the pattern cannot end with an escape character")
+	}
+
+	var sb strings.Builder
+
+	sb.WriteRune('^') // match start
+
+	for i, r := range pattern {
+		if r == LikeEscape { // if we find an escape sequence we just rewrite it
+			nextRune := pattern[i+1] // no illegal reference, because / can't be last
+
+			if nextRune == LikeAny || nextRune == LikeAll {
+				i += 1
```

Comment:
Use a normal loop if you modify i inside it. This doesn't work like you think it does. (i is by value in each loop iteration)